pkg/responsemeta/responsemeta.go

```diff
@@ -0,0 +1,111 @@
+package responsemeta
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	dispatch "github.com/authzed/spicedb/internal/proto/dispatch/v1"
+)
+
+// NewHandler creates a new response metadata handler for setting the response
+// metadata into the trailer of an API request.
+func NewHandler(ctx context.Context) *ResponseMetadataHandler {
```

Comment:
`NewUsageHandler` or `NewMeteringHandler`

Reply:
I dislike both names because they aren't being used for either system here; I originally had `NewResponseMetaHandler`, but then we have `responemeta.NewResponseMetaHandler`, which I thought was too verbose

Reply:
We decided that this either needed to be renamed or the fields made dynamic. I'm still in favor of it being renamed. `responsemeta.NewHandler` tells me nothing about what it's going to do.

Reply:
I'm open to other suggestions. `responsemeta.NewTrailerHandler`? `responsemeta.NewTrailerReporter`?